src/kixi/plugsocket.clj

```diff
@@ -1,25 +1,218 @@
+                          color java.awt.Color/BLACK
+                          border-size 2.0}}]
+  (.setBorderWidth cell TableCell$BorderEdge/bottom border-size)
+  (.setBorderColor cell TableCell$BorderEdge/bottom colour)
+  (.setBorderWidth cell TableCell$BorderEdge/top border-size)
+  (.setBorderColor cell TableCell$BorderEdge/top colour)
+  (.setBorderWidth cell TableCell$BorderEdge/left border-size)
+  (.setBorderColor cell TableCell$BorderEdge/left colour)
+  (.setBorderWidth cell TableCell$BorderEdge/right border-size)
+  (.setBorderColor cell TableCell$BorderEdge/right colour))
+
+(defn table-box [{:keys [ds slide x y colWidth
+                         border-colour
+                         border-color
+                         border-size
+                         font-size]
+                  :or {x false
+                       y false
+                       colWidth 200.0
+                       border-colour java.awt.Color/BLACK
+                       border-color java.awt.Color/BLACK
+                       border-size 2.0
+                       font-size 20.0}}]
+  (let [table (.createTable slide)
+        numColumns (tc/column-count ds)
+        numRows (tc/row-count ds)
+        headerRow (.addRow table)
+        ds-header (-> ds tc/column-names vec)]
+    (box-placement table x y false false)
+    (run! (fn [h]
+            (let [th (.addCell headerRow)
+                  p (.addNewTextParagraph th)
+                  r (.addNewTextRun p)]
+              (.setBold r true)
+              (.setTextAlign p TextParagraph$TextAlign/CENTER)
+              (.setText r (name (nth ds-header h)))
+              (.setFontSize r (+ font-size 10.0))
+              (.setColumnWidth table h (double colWidth))
+              (table-border {:cell th :colour border-color
+                             :border-size border-size})))
+          (range 0 numColumns))
+    (run! (fn [rowNum]
+            (let [tr (.addRow table)]
+              (run! (fn [h]
+                      (let [cell (.addCell tr)
+                            p (.addNewTextParagraph cell)
+                            r (.addNewTextRun p)
+                            col-name (nth ds-header h)]
+                        (.setText r (str (nth (ds col-name) rowNum)))
+                        (.setFontSize r font-size)
+                        (table-border {:cell cell :colour border-color
+                                       :border-size border-size})))
+                    (range 0 numColumns))))
+          (range 0 numRows))))
+
+(def slide-fns
+  {:text-box text-box
+   :chart-box chart-box
+   :image-box image-box
+   :table-box table-box})
+
+(defn create-slide
+  ;; takes a sequence of maps corresponding to a number of objects
+  ;; (text boxes, tables, images) to display on a slide
+  ([powerpoint]
+   (.createSlide powerpoint))
+  ([seq-of-maps powerpoint]
+   (create-slide seq-of-maps slide-fns powerpoint))
+  ([seq-of-maps config powerpoint]
```

Comment:
seq-of-maps could be called something like `slide-def` here, but I think this is fine.